service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerAboutToSubmitTest.java

```diff
@@ -0,0 +1,105 @@
+@WebMvcTest(HearingBookingDetailsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class HearingBookingDetailsControllerAboutToSubmitTest extends AbstractControllerTest {
+    private static final LocalDateTime TODAY = LocalDateTime.now();
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    HearingBookingDetailsControllerAboutToSubmitTest() {
+        super("add-hearing-bookings");
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenSubmittingEmptyEvent() {
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest(
+            Map.of(HEARING_DETAILS.getKey(), emptyList()), Map.of(HEARING_DETAILS.getKey(), emptyList())));
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(caseData.getHearingDetails()).isEmpty();
+    }
+
+    @Test
+    void shouldReturnHearingsWhenNoHearingsExistInPast() {
+        List<Element<HearingBooking>> hearingDetails = newArrayList(bookingWithStartDate(TODAY.plusDays(5)));
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest(
+            Map.of(HEARING_DETAILS.getKey(), hearingDetails), Map.of(HEARING_DETAILS.getKey(), hearingDetails)));
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(caseData.getHearingDetails()).hasSize(1);
+        assertThat(caseData.getHearingDetails()).isEqualTo(hearingDetails);
+    }
+
+    @Test
+    void shouldReturnHearingsWhenNoHearingsExistInFuture() {
+        List<Element<HearingBooking>> hearingDetails = newArrayList(bookingWithStartDate(TODAY.plusDays(-5)));
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest(
+            Map.of(HEARING_DETAILS.getKey(), emptyList()), Map.of(HEARING_DETAILS.getKey(), hearingDetails)));
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(caseData.getHearingDetails()).hasSize(1);
+        assertThat(caseData.getHearingDetails()).isEqualTo(hearingDetails);
+    }
+
+    @Test
+    void shouldReturnHearingsWhenHearingsInPastAndFutureExist() {
+        Element<HearingBooking> hearingDetail = bookingWithStartDate(TODAY.plusDays(5));
+        Element<HearingBooking> hearingDetailPast = bookingWithStartDate(TODAY.minusDays(5));
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToSubmitEvent(callbackRequest(
+            Map.of(HEARING_DETAILS.getKey(), newArrayList(hearingDetail)),
+            Map.of(HEARING_DETAILS.getKey(), newArrayList(hearingDetail, hearingDetailPast))));
+
+        CaseData caseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(caseData.getHearingDetails()).hasSize(2);
+        assertThat(caseData.getHearingDetails().get(0)).isEqualTo(hearingDetailPast);
+        assertThat(caseData.getHearingDetails().get(1)).isEqualTo(hearingDetail);
```

Comment:
assertThat(caseData.getHearingDetails()).containsExactly(hearingDetailPast, hearingDetail);

is enough, it assert order of elements